tests/mocked_plans/utils.py

```diff
@@ -0,0 +1,77 @@
+# Copyright 2023 Canonical Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Module to provide helper functions for writing mock upgrade tests."""
+from pathlib import Path
+from unittest.mock import AsyncMock, PropertyMock
+
+import yaml
+
+from cou.utils.juju_utils import Application, Machine, Model, Unit
+from tests.unit.utils import dedent_plan
+
+
+def get_sample_plan(source: Path) -> tuple[Model, str]:
```

Comment:
uber-nit: if you call the yaml file a sample plan, "getting it" would only mean reading it, and this function could perhaps more accurately be renamed to `parse_sample_plan`

Reply:
I like that, but I'll change it in another PR.

Reply:
reported #335